src/executors/StochasticNoiseSimulatorExecutor.cpp

```diff
@@ -0,0 +1,31 @@
+#include "executors/StochasticNoiseSimulatorExecutor.hpp"
+
+#include "StochasticNoiseSimulator.hpp"
+
+json StochasticNoiseSimulatorExecutor::execute(const SimulationTask& task) {
+  json       result;
+  auto const constructionStart = std::chrono::steady_clock::now();
+
+  auto qc = std::make_unique<qc::QuantumComputation>(task.getQc()->clone());
+  auto circuitSimulator =
+      std::make_unique<StochasticNoiseSimulator<>>(std::move(qc), 1, 1, 23);
+
+  auto const executionStart = std::chrono::steady_clock::now();
+
+  result["measurement_results"] = circuitSimulator->simulate(1024U);
+  // Add memory usage
+
+  auto const executionStop = std::chrono::steady_clock::now();
+  auto const constructionTime =
+      std::chrono::duration_cast<std::chrono::microseconds>(executionStart -
+                                                            constructionStart);
+  auto const execTime = std::chrono::duration_cast<std::chrono::microseconds>(
+      executionStop - executionStart);
+  result["construction_time"] = constructionTime.count();
+  result["execution_time"]    = execTime.count();
+
+  result["executor"] = getIdentifier();
+  result["task"]     = task.getIdentifier();
```

Comment:
https://github.com/tyi1025/dd_eval/pull/12/commits/1e86792f8bea7524f781c8f0a00104157f4618fa
Could you take a look at this commit and see if this way of refactoring is good? If so I'll do it for the rest of the executor classes

Reply:
That looks good, I think!
It should be possible to move the construction of the circuit to the base class though.
Since the SimulationExecutor is a template class. You will most likely have to define that in the header and make sure that you have the proper includes for all the classes that are used.